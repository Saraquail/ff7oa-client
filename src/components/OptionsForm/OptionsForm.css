form {
  display: flex;
  flex-direction: row;
  align-content: flex-start;
}

.sort-and-add {
  margin-top: 2em;
  margin-bottom: 2em;
}


input, select, button {
  width: 10em;
  align-self: center;
  font-size: 14px;
  margin-top: .5em;
}

#add-monster-button {
  width: 7em;
  font-size: 14px;
  margin-left: 1em;
}

select {
  background: #284883;
  color: whitesmoke;
  font-size: 14px;
}