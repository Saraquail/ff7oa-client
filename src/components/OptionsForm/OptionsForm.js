import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import './OptionsForm.css'

class OptionsForm extends Component {

  render () {
    return (
      <section id="bestiary-options">
      <label htmlFor="sort-by">Sort by:</label>
      <select defaultValue="" onChange={this.props.handleSort}>
        <option>-select-</option>
        <option>Name</option>
        <option>GIL</option>
        <option>Location</option>
        <option>EXP</option>
      </select>

      <form name="search-form" id="search-form" onSubmit={this.props.handleSearch}>
      <label htmlFor="filter-by">Search by:</label>
      <select defaultValue="" name="filter-by" id="filter-by">
        <option>-select-</option>
        <option>Location</option>
        <option>Name</option>
        <option>Drops</option>
        <option>Steal</option>
        <option>Enemy Skill</option>
      </select>
      <label htmlFor="search">Search:</label>
        <input type="text" name="search" id="search" />
        <button type='submit'>Search</button>
        <button onClick={this.props.handleReset}>Reset</button>
        </form>
      <button>
        <Link to="/monster-form">
          Add New
        </Link>
      </button>
    </section>
    )
  }
}

export default OptionsForm