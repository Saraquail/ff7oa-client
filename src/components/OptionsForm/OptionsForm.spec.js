import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';

import OptionsForm from './OptionsForm.js'

describe('<OptionsForm/>', () => {

  it('renders without crashing', () => {
  const div = document.createElement('div')

  ReactDOM.render(
  <BrowserRouter>
  <OptionsForm />
  </BrowserRouter>, div)

  ReactDOM.unmountComponentAtNode(div)
  })

  it('renders the UI as expected', () => {
    const tree = renderer
  .create(
    <BrowserRouter>
      <OptionsForm/>
    </BrowserRouter>
  )
  .toJSON();
    expect(tree).toMatchSnapshot();  
  })
})
