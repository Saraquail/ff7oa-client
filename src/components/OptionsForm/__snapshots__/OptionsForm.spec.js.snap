// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptionsForm/> renders the UI as expected 1`] = `
<section
  id="bestiary-options"
>
  <form
    id="search-form"
    name="search-form"
  >
    <label
      htmlFor="filter-by"
    >
      Search by:
    </label>
    <select
      defaultValue=""
      id="filter-by"
      name="filter-by"
    >
      <option>
        -select-
      </option>
      <option>
        Location
      </option>
      <option>
        Name
      </option>
      <option>
        Drops
      </option>
      <option>
        Steal
      </option>
      <option>
        Enemy Skill
      </option>
    </select>
    <label
      htmlFor="search"
    >
      Search:
    </label>
    <input
      id="search"
      name="search"
      type="text"
    />
    <button
      type="submit"
    >
      Search
    </button>
    <button>
      Reset
    </button>
  </form>
  <div
    className="sort-and-add"
  >
    <label
      htmlFor="sort-by"
    >
      Sort by:
    </label>
    <select
      defaultValue=""
      id="sort-by"
      name="sort-by"
    >
      <option>
        -select-
      </option>
      <option>
        Name
      </option>
      <option>
        GIL
      </option>
      <option>
        Location
      </option>
      <option>
        EXP
      </option>
    </select>
    <a
      href="/monster-form"
      id="add-monster-button"
      onClick={[Function]}
    >
      <button
        id="add-monster-button"
      >
        Add New
      </button>
    </a>
  </div>
</section>
`;
